operator/sidecar/sidecar.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sidecar
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/breaker"
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	libclient "github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/client/identityfile"
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+const (
+	DefaultLocalAddr  = "127.0.0.1:3025"
+	DefaultConfigPath = "/etc/teleport/teleport.yaml"
+	DefaultDataDir    = "/var/lib/teleport"
+	DefaultUser       = "teleport-operator-sidecar"
+	DefaultRole       = "teleport-operator-sidecar"
+)
+
+// Options configure the sidecar connection.
+type Options struct {
+	// ConfigPath is a path to the Teleport configuration file e.g. /etc/teleport/teleport.yaml.
+	ConfigPath string
+
+	// DataDir is a path to the Teleport data dir e.g. /var/lib/teleport.
+	DataDir string
+
+	// Addr is an endpoint of Teleport e.g. 127.0.0.1:3025.
+	Addr string
+
+	// User is a user used to access Teleport Auth/Proxy/Tunnel server.
+	User string
+
+	// Role is a role allowed to manage Teleport resources.
+	Role string
+}
+
+func writeIdentityFile(ctx context.Context, clusterAPI auth.ClientI, identityFilePath, userName string) error {
+	// generate a keypair:
+	key, err := libclient.NewKey()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	reqExpiry := time.Now().UTC().Add(1 * time.Hour)
```

Comment:
Sorry, I think we have to change this
It's basically creating a certificate that allows our client to access the API for 1 hour, which will not be enough
We have to either increase the expiration time or the client must be able to re-create a certificate when the one we have expires

Reply:
This has been addressed in https://github.com/gravitational/teleport/pull/13331/commits/0fa831199553e0dec95075bf4ef417ea8f934a21

This is a pretty big change. tl;dr
- Local auth tsh-style connection happens on startup to create roles
- We wait for leader election
- When leader it registers a new teleport bot (bot currently don't support multiple tokens on the same bot, we have de delete/recreate)
- When teleport bort is registered we start a `tbot`, it renews its own cert and the client one
- For each request we create a new client using the certs from the tbot memory. This is a bit ineffective, at some point if tbot exposes a renewal channel we can listen on this should be improved.